test/tests/kernels/conservative_advection/tests

```diff
@@ -63,12 +70,25 @@
     [no_upwinding_1D_mat]
       type = Exodiff
       input = no_upwinding_1D.i
-      prereq = upwinding/no_upwinding_1D
       exodiff = no_upwinding_1D_mat_out.e
       abs_zero = 1E-7
-      cli_args = "Outputs/file_base=no_upwinding_1D_mat_out Kernels/advection/velocity_material=v Materials/v/type=GenericConstantVectorMaterial Materials/v/prop_names=v Materials/v/prop_values='1 0 0'"
+      cli_args = "Outputs/file_base=no_upwinding_1D_mat_out Kernels/advection/velocity_material=v"
       detail = 'without upwinding in 1D, with velocity given as a material,'
     []
+    [no_upwinding_1D_coupled_aux]
+      type = Exodiff
+      input = no_upwinding_1D_coupled_aux.i
+      exodiff = no_upwinding_1D_coupled_aux_out.e
+      abs_zero = 1E-7
+      detail = 'without upwinding in 1D, with velocity taken as the gradient of a coupled aux variable,'
+    []
+    [no_upwinding_1D_coupled]
+      type = Exodiff
```

Comment:
I'd like to see a `PetscJacobianTester` for the nonlinear variable coupling case as well